### PR TITLE
add alpine dockerfile

### PR DESCRIPTION
**Pull request by knowname**
# Issue

https://github.com/Project-OSRM/osrm-backend/issues/6956

## Tasklist

 - [ ] CHANGELOG.md entry ([How to write a changelog entry](http://keepachangelog.com/en/1.0.0/#how))
 - [ ] update relevant [Wiki pages](https://github.com/Project-OSRM/osrm-backend/wiki)
 - [ ] add tests (see [testing documentation](https://github.com/Project-OSRM/osrm-backend/blob/master/docs/testing.md))
 - [ ] review
 - [ ] adjust for comments
 - [ ] cherry pick to release branch

## Requirements / Relations

None

### CODE REVIEW

File: docker/hooks/build
Comment (SiarheiFedartsou) [inline docker/hooks/build:13]: Actually this file is not used anymore (at least as far as I understand). 

During release we build images without using it, see: https://github.com/Project-OSRM/osrm-backend/blob/7ffc08be2849742ceb92fbdc756b7af3ba6f8f55/.github/workflows/osrm-backend-docker.yml#L56

Also we have this for CI: https://github.com/Project-OSRM/osrm-backend/blob/7ffc08be2849742ceb92fbdc756b7af3ba6f8f55/.github/workflows/osrm-backend.yml#L119


In general IMO your PR is already very welcome and I am ready to merge it, but if you have some time would be great to also update these CI jobs (may be at least [this one](https://github.com/Project-OSRM/osrm-backend/blob/7ffc08be2849742ceb92fbdc756b7af3ba6f8f55/.github/workflows/osrm-backend.yml#L119) to make sure that this Alpine image is buildable after each PR).

Comment (knowname) [reply docker/hooks/build:13]: It's handy to have something around to remind you how to build it locally.
Maybe thats the reason why it's still around?

👍 I'll look into the CI jobs. Only built it locally so far.

Comment (SiarheiFedartsou) [reply docker/hooks/build:13]: It seems until you merge at least one PR you will have to wait for my approval before running CI, so it will make feedback loop for you quite long... We can merge Dockerfile only first (and then add CI in separate PR) if it will be a problem - just let me know.

Comment (knowname) [reply docker/hooks/build:13]: Fine with me. I just removed the last commit with the workflow changes.

**Review by SiarheiFedartsou — APPROVED** [review]

**Pull request merged**
